source/MaterialXGenShader/ShaderGenerator.h

```diff
@@ -130,6 +133,20 @@ class ShaderGenerator
     /// shader generators node context storage and returned.
     GenContextPtr createContext(int id);
 
+    /// Utility to emit a block of either uniform or constant variables
+    /// @param block Block to emit.
+    /// @param asConstant If set to "true", will emit a block constants as opposed to a block of uniforms. 
+    /// The differense is that a constant qualifier is used versus a unifork qualifier. Qualifiers are specified by the 
+    /// syntax for the generator.
+    /// @param shader Shader to emit to.
+    virtual void emitVariableBlock(const Shader::VariableBlock& block, bool asConstant, Shader& shader);
```

Comment:
Could the interface on `emitVariableBlock `be made the same as `emitVariable`: using `qualifier `instead of `asConstant`, for consistency.